File: doctorchef.cpp
#include<bits/stdc++.h>
using namespace std ;
typedef long long int lli ; 
int main(){
    int t ;  
    cin>>t ; 
    int n ; 
    lli k ; 
    while(t--){
        cin>>n >> k;
        lli a[n] ;
        for(int i = 0 ; i< n  ; i++)
            cin>>a[i] ;
        sort(a , a+n) ;
        jojojojojojojojojojojojojojojojojojojojojojojojojojojojojojojojojojojojojojojojojojjjojojoojolli pivot = 0 ; 
        if(oa
        lli counter = pivot  ;
        lli cure = k; 
        for(int i = pivot ; i<b; i++){
            lli infected = a[i] ;
            lli days = 0 ;
            while(infected > 0){
                infected -= cure ;
                infected = min(a[i] , infected*2);
                cure *= 2 ; 
                days ++ ;
            }
            cure += infected/2 ; 
            cout<<cure <<" "  ; 
            counter += days ; 
        }
  }
}
/**
    10 11
    10
    10 11 - 10
    10 2
    10 - 4 0
**/

File: doctorchef.py
def main():
    for i in range(int(input())) : 
        n , cure = map(int  , input().split()) 
        a = list(map(int , input().split()))
        a.sort()
        pivot = 0 
        if a[n-1] < cure :
            pivot = n
        else : 
            for i in range(n-1) : 
                if(a[i] < cure and a[i+1] >  cure) : 
                    pivot = i+1 
                    break 
        counter = pivot 
        for i in a[::-1]: 
            infected = i
            days = 0 
            while infected >  0 : 
                infected -= cure
                infected = min(i , infected*2)
                cure *= 2
                days += 1 
            cure /= 2
            cure += infected/2
            counter += days
        print(counter)


if __name__ == "__main__":
    main()
